Classes/git/PBGitRepository.m

```diff
@@ -891,6 +891,31 @@ - (BOOL) cherryPickRefish:(id <PBGitRefish>)ref error:(NSError **)error
 	return YES;
 }
 
+- (BOOL) resetSoftRefish:(id <PBGitRefish>)ref error:(NSError **)error
```

Comment:
I'd prefer this method to be more "generic" since it's on `PBGitRepository`.

So maybe `- (BOOL)reset:(PBResetMode)mode to:(id <PBGitRefish>)ref error:(NSError **)error` ? Alternatively, feel free to reuse the Objective-Git enum for the reset mode.